scripts/xtensa-build-zephyr.py

```diff
@@ -209,7 +212,8 @@ def parse_args():
 	if len(sys.argv) == 1:
 			parser.print_help()
 			sys.exit(0)
-
+	if args.cavs_fw_style and args.ipc != "IPC4":
+		raise RuntimeError(f"Argument --cavs-fw-style without -i IPC4 makes no sense")
```

Comment:
Could you downgrade this to a `warnings.warn()`? I agree it makes no sense but if someone does this then maybe they may have some strange reasons to do it, I mean it's not something that is likely to happen by accident.

Reply:
Sounds good since this provides more flexibility to any script users, will refine it

Reply:
Also I feel like https://github.com/thesofproject/sof/pull/5959/files#diff-daebe2b740a448f2617ecb57b6bb2ac998b0ba033b67552559b5b1344f0ee1deR519 this condition should be extracted out of the IPC4 condition if we want to make them independent to provide more building flexibility

Reply:
I don't think we should compromise here,  the probability of this usage is ZERO.  if ipc=IPC3 and cavs_fw_style=true, how should we organize the file structure？

Reply:
> if ipc=IPC3 and cavs_fw_style=true, how should we organize the file structure？

Ideally, these two options should be implemented independently of each other.

If not possible then raise an exception at the exact code location(s) where it's not possible and causing an implementation issue.

Either way don't implement anything at argument parsing time (= in a completely different place in the code). 

It is a build script for CI and developers; people are supposed to read the corresponding `--help` text before using any new `--option`, they are supposed to understand their purpose and what combinations make sense versus not. As long as the default values do something simple and sensible with good logs then no extra code and no hand holding required.

This script is already too big; a lot of it should be transferred elsewhere (especially to CMake). I digress sorry.